src/dbnode/storage/index/options.go

```diff
@@ -414,3 +416,13 @@ func (o *opts) SetMmapReporter(mmapReporter mmap.Reporter) Options {
 func (o *opts) MmapReporter() mmap.Reporter {
 	return o.mmapReporter
 }
+
+func (o *opts) SetQueryStats(value stats.QueryStats) Options {
+	opts := *o
+	opts.queryStats = value
```

Comment:
Can you add to the `func (o *opts) Validate() error` method in this file a check for `if o.queryStats == nil { return errNoQueryStats }` and define `errNoQueryStats` alongside the other err messages?